### PR TITLE
417 noxgenerator exclude soft delete entities for queries

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

**Comment by anna-naumova25** [comment]
@rochar I've just thought about adding IsDeleted property to IDeleteEntity. Maybe it would be a cleaner solution. Please let me know your thoughts.

File: src/Nox.Generator/Application/Queries/ByIdQueryGenerator.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public static void Generate(SourceProductionContext context, NoxSolutionCodeGene
                 .WithClassName($"Get{entity.Name}ByIdQuery")
                 .WithFileNamePrefix($"Queries")
                 .WithObject("entity", entity)
+                .WithObject("isVersioned", (entity.Persistence?.IsVersioned ?? true))
```
Comment (rochar) [inline src/Nox.Generator/Application/Queries/ByIdQueryGenerator.cs:27]: I know we have this on other generator, we would prefer not to have .WithObject and make it clear and obvious on the template {{if entity.Persistence.IsVersioned}}

File: src/Nox.Generator/Application/Queries/Query.template.cs [hunk 2]
```diff
@@ -21,6 +21,7 @@ public class Get{{entity.PluralName}}QueryHandler : IRequestHandler<Get{{entity.
 
     public Task<IQueryable<O{{entity.Name}}>> Handle(Get{{entity.PluralName}}Query request, CancellationToken cancellationToken)
     {
-        return Task.FromResult((IQueryable<O{{entity.Name}}>)DataDbContext.{{entity.PluralName}});
+        var item = (IQueryable<O{{entity.Name}}>)DataDbContext.{{entity.PluralName}}{{if isVersioned}}.Where(r => !(r.Deleted == true)){{end}};
```
Comment (rochar) [inline src/Nox.Generator/Application/Queries/Query.template.cs:24]: do we need r.Deleted == true? r.Deleted should be enough?

File: src/Nox.Generator/Application/Queries/QueryGenerator.cs [hunk 3]
```diff
@@ -25,6 +25,7 @@ public static void Generate(SourceProductionContext context, NoxSolutionCodeGene
                 .WithClassName($"Get{entity.PluralName}Queries")
                 .WithFileNamePrefix($"Queries")
                 .WithObject("entity", entity)
+                .WithObject("isVersioned", (entity.Persistence?.IsVersioned ?? true))
```
Comment (rochar) [inline src/Nox.Generator/Application/Queries/QueryGenerator.cs:28]: same has previous comment

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, just two small comments to review

File: src/Nox.Generator/Application/Queries/Query.template.cs (hunk 2 above)
Comment (anna-naumova25) [reply src/Nox.Generator/Application/Queries/Query.template.cs:24]: r.Deleted won't work as Deleted is a nullable field and it cannot automatically convert bool? to bool.

File: src/Nox.Generator/Application/Queries/ByIdQueryGenerator.cs (hunk 1 above)
Comment (anna-naumova25) [reply src/Nox.Generator/Application/Queries/ByIdQueryGenerator.cs]: Updated

**Review by rochar — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**